File: src/main/java/net/ndrei/teslacorelib/netsync/TeslaCorePackets.java
package net.ndrei.teslacorelib.netsync;

import net.minecraftforge.fml.common.network.NetworkRegistry;
import net.minecraftforge.fml.common.network.simpleimpl.IMessage;
import net.minecraftforge.fml.common.network.simpleimpl.SimpleNetworkWrapper;
import net.minecraftforge.fml.relauncher.Side;

/**
 * Created by CF on 2016-12-03.
 */
@SuppressWarnings("unused")
public class TeslaCorePackets implements ITeslaCorePackets {
    private SimpleNetworkWrapper wrapper;

    public TeslaCorePackets(String channelName) {
        this.wrapper = new SimpleNetworkWrapper(channelName);

        this.wrapper.registerMessage(SimpleNBTHandler.class, SimpleNBTMessage.class, 1, Side.CLIENT);
        this.wrapper.registerMessage(SimpleNBTHandler.class, SimpleNBTMessage.class, 1, Side.SERVER);
    }

    @Override
    public void send(IMessage message) {
        NetworkRegistry.TargetPoint target = null;
        if (message instanceof SimpleNBTMessage) {
            target = ((SimpleNBTMessage)message).getTargetPoint();
        }

        if (target == null) {
            this.wrapper.sendToAll(message);
        }
        else {
            this.wrapper.sendToAllAround(message, target);
        }
    }

    @Override
    public void sendToServer(IMessage message) {
        this.wrapper.sendToServer(message);
    }
}

File: src/main/java/net/ndrei/teslacorelib/netsync/TeslaCorePackets.kt
package net.ndrei.teslacorelib.netsync

import net.minecraftforge.fml.common.network.simpleimpl.IMessage
import net.minecraftforge.fml.common.network.simpleimpl.SimpleNetworkWrapper
import net.minecraftforge.fml.relauncher.Side

/**
 * Created by CF on 2017-06-28.
 */
class TeslaCorePackets(channelName: String) : ITeslaCorePackets {
    private val wrapper: SimpleNetworkWrapper = SimpleNetworkWrapper(channelName)

    init {
        this.wrapper.registerMessage(SimpleNBTHandler::class.java, SimpleNBTMessage::class.java, 1, Side.CLIENT)
        this.wrapper.registerMessage(SimpleNBTHandler::class.java, SimpleNBTMessage::class.java, 1, Side.SERVER)
    }

    override fun send(message: IMessage) {
        if (message is SimpleNBTMessage) {
            this.wrapper.sendToAllAround(message, message.targetPoint)
        }
        else {
            this.wrapper.sendToAll(message)
        }
    }

    override fun sendToServer(message: IMessage) {
        this.wrapper.sendToServer(message)
    }
}
